subsys/net/lib/nrf_cloud/src/nrf_cloud_agps.c

```diff
@@ -0,0 +1,561 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <drivers/gps.h>
+#include <net/socket.h>
+#include <nrf_socket.h>
+
+#include <modem_info.h>
+#include <nrf_cloud_agps.h>
+#include <nrf_cloud_transport.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(nrf_cloud_agps, CONFIG_NRF_CLOUD_AGPS_LOG_LEVEL);
+
+#include "nrf_cloud_agps_schema_v1.h"
+
+extern void agps_print(enum nrf_cloud_agps_type type, void *data);
+
+static int fd = -1;
+static bool agps_print_enabled;
+static struct device *gps_dev;
+
+static enum gps_agps_type type_lookup_socket2gps[] = {
+	[NRF_GNSS_AGPS_UTC_PARAMETERS]	= GPS_AGPS_UTC_PARAMETERS,
+	[NRF_GNSS_AGPS_EPHEMERIDES]	= GPS_AGPS_EPHEMERIDES,
+	[NRF_GNSS_AGPS_ALMANAC]		= GPS_AGPS_ALMANAC,
+	[NRF_GNSS_AGPS_KLOBUCHAR_IONOSPHERIC_CORRECTION]
+					= GPS_AGPS_KLOBUCHAR_CORRECTION,
+	[NRF_GNSS_AGPS_NEQUICK_IONOSPHERIC_CORRECTION]
+					= GPS_AGPS_NEQUICK_CORRECTION,
+	[NRF_GNSS_AGPS_GPS_SYSTEM_CLOCK_AND_TOWS]
+					= GPS_AGPS_GPS_SYSTEM_CLOCK_AND_TOWS,
+	[NRF_GNSS_AGPS_LOCATION]	= GPS_AGPS_LOCATION,
+	[NRF_GNSS_AGPS_INTEGRITY]	= GPS_AGPS_INTEGRITY,
+};
+
+void agps_print_enable(bool enable)
```

Comment:
Is this function actually used anywhere?

Reply:
It's used in a test app that I made to be able to see all A-GPS data. It won't go public, so I could probably remove it, along with `nrf_cloud_agps_utils.c`?

Reply:
Well, I'm not sure if we should remove it, it seems useful. Perhaps we should add a Kconfig switch to the library to enable debug mode?